### PR TITLE
Refactor Python requirements

### PR DESCRIPTION
**Pull request by jezdez**

### CODE REVIEW

**Comment by jezdez** [comment]
@robhudson r?

**Comment by groovecoder** [comment]
giphy

**Comment by jezdez** [comment]
anvo1xkaa18mq

**Comment by jezdez** [comment]
zbknesihcsdvg

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,3 @@
-branches:
-    only:
-        - master
```
Comment (robhudson) [inline .travis.yml]: We want this forever?

File: docs/development.rst [hunk 2]
```diff
@@ -97,30 +97,75 @@ Tests
...
 
-Compiled Packages
-~~~~~~~~~~~~~~~~~
+Here's what that folder contains:
 
-There are a small number of compiled packages, including the MySQL Python
-client. You can install these using ``pip`` or via a package manager.
-To use ``pip``, you only need to do the following.
+- ``compiled.txt`` - contains dependencies that require a compiler and may
+  need to be treated differently dependending on environment
```
Comment (robhudson) [inline docs/development.rst]: depending

File: docs/development.rst [hunk 3]
```diff
@@ -97,30 +97,75 @@ Tests
...
+
+Updating a requirement
+~~~~~~~~~~~~~~~~~~~~~~
+
+Follow the same steps as when adding a requirement but replace the old peep
+hash in the requirements file. Don't forget to run afterwards::
 
-    deactivate
-    sudo pip install -r requirements/compiled.txt
+    script/peep.py install -r requirements/default.txt
```
Comment (robhudson) [inline docs/development.rst]: Thanks for documenting all that. :+1:

File: docs/development.rst [hunk 4]
```diff
@@ -97,30 +97,75 @@ Tests
...
+
+    # sha256: paFCZIUSX_kQWjcNx9em6npTILXRgCcjA9QppD-BL-U
+
+Add this string above the line of the requirement string in the requirements
+file, e.g.::
+
+    # sha256: paFCZIUSX_kQWjcNx9em6npTILXRgCcjA9QppD-BL-U
+    django-pipeline==1.6.0
+
+Then verify if the hash stil matches and install the new dependency in the VM::
```
Comment (robhudson) [inline docs/development.rst]: still

File: settings.py [hunk 5]
```diff
@@ -458,6 +458,7 @@ def get_locales():
     'dbgettext',
     'honeypot',
     'cacheback',
+    'django_extensions',
```
Comment (robhudson) [inline settings.py]: Why do we need this?

Comment (robhudson) [inline settings.py]: I'm actually not very familiar with what's in this package that we might use.

**Comment by robhudson** [comment]
Commented on some typos and curious about a few things. r+wc.

File: .travis.yml (hunk 1 above)
Comment (jezdez) [inline .travis.yml]: Hm, I'm not sure, @groovecoder I think you add this. Do we want to keep this?

File: docs/development.rst (hunk 3 above)
Comment (jezdez) [inline docs/development.rst]: :scroll:

File: settings.py (hunk 5 above)
Comment (jezdez) [inline settings.py]: It adds a few useful management commands, and provides a model field for easy timespamps. It's well maintained and really worth the addition. We previously had it in settings_local.py but there was no reason to not have it here either.

File: docs/development.rst (hunk 2 above)
Comment (jezdez) [inline docs/development.rst]: :+1:

File: docs/development.rst (hunk 4 above)
Comment (jezdez) [inline docs/development.rst]: :+1:

**Comment by jezdez** [comment]
vjnwqeukurfmw

**Pull request merged**